### DISCUSSION TITLE
Unexpected event performence

### DISCUSSION THREAD

**Discussion by NotAlwaysTrue** [FakeFishGames/Barotrauma#14264] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

While trigger following event with consle command triggerevent will lead to unexpected performence
```
<RandomEvents>
  <EventPrefabs>
    <ScriptedEvent identifier="missionevent">
      <Label name="start" />
      <TagAction criteria="players" tag="enteredplayer" submarinetype="Outpost" continueifnotargetsfound="true" />
      <AfflictionAction affliction="someaffliction" strength="20" targettag="enteredplayer" />
      <ClearTagAction tag="enteredplayer" />
      <WaitAction time="3" />
      <GoTo name="start" />
    </ScriptedEvent>
  </EventPrefabs>
</RandomEvents>
``` 
The code indicated only players inside the outpost will be tagged with 'enteredplayer' and receive damage. However, a player in a sub can also be affected.(MP and Local subeditor)

A different phenomenon were found while using the built-in eventeditor. In the testing outpost, only the first character you control will receive the damage, changing to other characters and enter the outpost will not receive any damage.

If the code is wrong, please point it out. I am expecting a ScriptedEvent that everyone is safe while inside the sub and ONLY the person who entered the outpost will receive affliction and stop receiving affliction while he has left it

### Reproduction steps

Local Subeditor:
1. Create a event the same as pervious posted code.
2. Goto subeditor, open whatever sub you want
3. Use `triggerevent` command to trigger it
4. Found your character receiving unexpected damage.

MP:
1. Create a event the same as pervious posted code.
2. Start a MP server
3. Start a campaign
5. Stay in the sub
6. Use `triggerevent` command to trigger it
7. Found your character receiving unexpected damage.

Eventeditor:
1. Create a event the same as pervious posted code.
2. Save and enter the testing outpost
3. Watching captain(spawn in a outpost) receiving damage as expected
4. Switch to another character and enter the outpost
5. Found the new character didn't receive damage.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Other

### -

MP,SP,Subeditor and Eventeditor

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by A2ur3** [comment]
The TagAction here doesn't seems work for some reason
